contracts/bridge/Bridge.sol

```diff
@@ -122,7 +126,8 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
     {
         require(handleNonce == _requestNonce, "mismatched handle / request nonce");
 
-        emit HandleValueTransfer(TokenType.ERC20, _from, _to, _tokenAddress, _value, handleNonce);
+        emit HandleValueTransfer(_requestTxHash, TokenType.ERC20, _from, _to, _tokenAddress, _value, handleNonce);
+        _setHandledRequestTxHash(_requestTxHash);
```

Comment:
How about switching the order of L129 and L130?
Just to ensure an event has occurred after a value set. 
(And, the same comment for L156, L180)

Reply:
Solidity event will be canceled if it is reverted. :)